### PR TITLE
Removes CCIA

### PR DESCRIPTION
**Pull request by Arrow768**
view duty log verb and changes the way their recorder works.

CCIA recordings are now written to the database once the interview is complete (if enabled) and printed out.

### CODE REVIEW

**Comment by ben10083** [comment]
Thank god, I hate that staff team

**Comment by skull132** [comment]
Probably need to update the migration.

File: code/modules/cciaa/cciaa_items.dm [hunk 1]
```diff
@@ -37,68 +40,11 @@
...
-
 /obj/item/device/taperecorder/cciaa/record()
 	set name = "Start Recording"
 	set category = "Recorder"
 
-	if(!usr.client.holder && !(usr.client.holder.rights & R_CCIAA))
+	if(!check_rights(R_CCIAA,FALSE))
 		to_chat(usr, "<span class='notice'>The device beeps and flashes \"Unauthorised user.\".</span>")
 		return
 	if(usr.stat)
```
Comment (skull132) [inline code/modules/cciaa/cciaa_items.dm]:
```suggestion
	if(use_check_and_message(usr))
```

File: code/modules/cciaa/cciaa_items.dm [hunk 2]
```diff
@@ -107,88 +53,109 @@
...
-		to_chat(usr, "<span class='notice'>The device beeps and flashes \"No data entered, Aborting\".</span>")
+	//If nothing has been done with the device yet
+	if(!report_id && !interviewee_id)
+		if(config.sql_ccia_logs)
+			//Get the active cases from the database and display them
+			var/list/reports = list()
+			var/DBQuery/report_query = dbcon.NewQuery("SELECT id, report_date, title FROM ss13_ccia_reports WHERE status = 'approved' AND deleted_at IS NULL")
+			report_query.Execute()
+			while(report_query.NextRow())
+				reports["[report_query.item[2]] - [report_query.item[3]]"] = "[report_query.item[1]]"
```
Comment (skull132) [inline code/modules/cciaa/cciaa_items.dm]: This basically pulls all. Might want to pagify or, at minimum, tick-check this with `CHECK_TICK`

File: code/modules/cciaa/cciaa_items.dm [hunk 3]
```diff
@@ -107,88 +53,109 @@
...
+		to_chat(usr, "<span class='notice'>The device beeps and flashes \"Writing to [fileName]\".</span>")
 
-	return
+		return
 
 /obj/item/device/taperecorder/cciaa/stop()
 	set name = "Stop"
 	set category = "Recorder"
 
 	if(usr.stat)
```
Comment (skull132) [inline code/modules/cciaa/cciaa_items.dm]:
```suggestion
	if(use_check_and_message(usr))
```

File: code/modules/cciaa/cciaa_items.dm [hunk 4]
```diff
@@ -198,14 +165,30 @@
...
 	var/info = "[dat]"
 	P.set_content_unsafe(pname, info)
-	P.forceMove(get_turf(src.loc))
+	return P
 
+/obj/item/device/taperecorder/cciaa/print_transcript()
+	set name = "Print Transcript"
+	set category = "Recorder"
+
+	if(usr.stat)
```
Comment (skull132) [inline code/modules/cciaa/cciaa_items.dm]:
```suggestion
	if(use_check_and_message(usr))
```

File: code/modules/cciaa/cciaa_items.dm [hunk 5]
```diff
@@ -214,8 +197,8 @@
 
 	if(usr.stat)
```
Comment (skull132) [inline code/modules/cciaa/cciaa_items.dm]:
```suggestion
	if(use_check_and_message(usr))
```

File: code/modules/cciaa/cciaa_items.dm [hunk 6]
```diff
@@ -225,16 +208,37 @@
...
-	return
+	if(!report_id)
+		record()
+		return
+
+	if(interviewee_id)
+		to_chat(user,"<span class='notice'>The device beeps and flashes \"A interviewee has already been associated with this interview\".</span>")
+		return
+	
+	if(iscarbon(user))
```
Comment (skull132) [inline code/modules/cciaa/cciaa_items.dm]:
```suggestion
	if(ishuman(user))
```

File: code/modules/cciaa/cciaa_items.dm (hunk 2 above)
Comment (Arrow768) [reply code/modules/cciaa/cciaa_items.dm]: Well, that pulls all active IRs, so that are around 5-10 at most

**Pull request merged**